tests/test_shadows.py

```diff
@@ -0,0 +1,95 @@
+import numpy as np
+import pytest
+from pytest_lazyfixture import lazy_fixture as lf
+import tensorcircuit as tc
+from tensorcircuit.shadows import shadow_snapshots, local_snapshot_states, global_shadow_state, global_shadow_state1, \
+    global_shadow_state2, entropy_shadow, expection_ps_shadow, shadow_bound
+
+
+@pytest.mark.parametrize("backend", [lf("tfb"), lf("jaxb")])
+def test_jit(backend):
+    nq, ns = 8, 10000
+
+    c = tc.Circuit(nq)
+    for i in range(nq):
+        c.H(i)
+    for i in range(0, 2):
+        for j in range(nq):
+            c.cnot(j, (j + 1) % nq)
+        for j in range(nq):
+            c.rx(j, theta=0.5 * np.pi)
+
+    def classical_shadow(psi, pauli_strings, status, sub, x, y, z):
+        snapshots = shadow_snapshots(psi, pauli_strings, status)
+        lss_states = local_snapshot_states(snapshots, pauli_strings)
+        sdw_state = global_shadow_state(lss_states, sub=sub)
```

Comment:
could also add a test to assert close between global state from local states and snapshots directly